server/events/project_command_builder_test.go

```diff
@@ -716,3 +716,95 @@ func TestDefaultProjectCommandBuilder_EscapeArgs(t *testing.T) {
 		})
 	}
 }
+
+// Test that terraform version is used when specified in terraform configuration
+func TestDefaultProjectCommandBuilder_TerraformVersion(t *testing.T) {
+	atlantisYamlContent := `
+version: 3
+projects:
+- dir: project1 # project1 uses the defaults
+  terraform_version: v0.12.6
+`
+	versionConfig := `
+terraform {
+  required_version = "= 0.12.8"
+}	  
+`
+	testCases := map[string]struct {
+		DirStructure  map[string]interface{}
+		AtlantisYAML  string
+		ModifiedFiles []string
+		Exp           map[string]interface{}
+	}{
+		"no atlantis.yaml": {
+			DirStructure: map[string]interface{}{
+				"project1": map[string]interface{}{
+					"main.tf":    nil,
+					"version.tf": versionConfig,
+				},
+			},
+			ModifiedFiles: []string{"project1/main.tf"},
+			Exp: map[string]interface{}{
+				"TerraformVersion": []int{0, 12, 8},
+			},
+		},
+		// atlantis.yaml should take precedence over terraform config
+		"defined in atlantis.yaml": {
+			DirStructure: map[string]interface{}{
+				"project1": map[string]interface{}{
+					"main.tf":    nil,
+					"version.tf": versionConfig,
+				},
+				yaml.AtlantisYAMLFilename: atlantisYamlContent,
+			},
+			ModifiedFiles: []string{"project1/main.tf"},
+			Exp: map[string]interface{}{
+				"TerraformVersion": []int{0, 12, 6},
+			},
+		},
```

Comment:
can you also test:
* when using a non-exact version, ex. `> 0.12.8`
* when there's only project config
* when there's no project config or required version

Reply:
Wilco

Reply:
Done